### PR TITLE
Add User Review Functionality

### PR DESCRIPTION
**Pull request by akmatchev**
## Overview

Added the User Review Model with the basic CRUD endpoints. 

## Changes Made

I create new files in the models, repositories, services, and controllers to create the User Review model. A new file in migrations was also created to account for a database migration since I created two new columns in the user model. User Reviews has the basic CRUD endpoints except for updating the User Review, but this should be pretty irrelevant since on Uber for example, I don't think you can update the review after you review the driver. I also added two attributes to the user model: stars and num reviews. "Stars" represents the sum of the number of stars for all the reviews, and "numReviews" represents the number of reviews a seller has gotten. Thus, the average number of stars can be determined by dividing "stars" by "numReviews". 

## Test Coverage

Wrote test cases in the file.

### CODE REVIEW

File: src/api/controllers/UserReviewController.ts [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+    }
+
+    @Get()
+    async getUserReviews(): Promise<GetUserReviewsResponse> {
+        return { userReviews: await this.userReviewService.getAllUserReviews() };
+    }
+
+    @Get('id/:id/')
+    async getUserReviewsById(@Params() params: UuidParam): Promise<GetUserReviewResponse> {
+        return { userReview: await this.userReviewService.getUserReviewById(params)}
```
Comment (snajima) [inline src/api/controllers/UserReviewController.ts:26]: space before the ending curly brace and semicolon at the end of this line - let's keep styling consistent

File: src/types/ApiResponses.ts [hunk 2]
```diff
@@ -129,6 +130,25 @@ export interface GetRequestResponse {
...
+// USER REVIEWS
+
+export interface UserReview {
+    id: Uuid,
+    fulfilled: boolean,
+    stars: number,
+    comments: string,
+    date: Date,
+    buyer: PrivateProfile,
+    seller: PrivateProfile
```
Comment (snajima) [inline src/types/ApiResponses.ts:142]: comma on this line too

File: src/types/ApiRequests.ts [hunk 3]
```diff
@@ -102,6 +102,15 @@ export interface CreateRequestRequest {
...
 }
 
+// USER REVIEW
+
+export interface CreateUserReviewRequest {
+    fulfilled: boolean,
+    stars: number,
+    comments: string,
+    buyerId: Uuid
+    sellerId: Uuid
```
Comment (snajima) [inline src/types/ApiRequests.ts:112]: comma on both lines

File: src/models/UserReviewModel.ts [hunk 4]
```diff
@@ -0,0 +1,43 @@
...
+
+    public getReviewInfo(): UserReview {
+        return {
+            id: this.id,
+            fulfilled: this.fulfilled,
+            stars: this.stars,
+            comments: this.comments,
+            date: this.date,
+            buyer: this.buyer.getUserProfile(),
+            seller: this.seller.getUserProfile()
```
Comment (snajima) [inline src/models/UserReviewModel.ts:40]: comma on this line

**Review by snajima — APPROVED** [review]
lgtm, just a few stylistic changes + let's add test cases if we can

**Review by snajima — DISMISSED** [review]
sorry i meant to request changes lol

File: src/repositories/PostRepository.ts [hunk 5]
```diff
@@ -95,8 +95,8 @@ export class PostRepository extends AbstractRepository<PostModel> {
     return await this.repository
       .createQueryBuilder("post")
       .leftJoinAndSelect("post.user", "user")
-      .where("post.price >= :lowerBound", {lowerBound: lowerBound})
-      .andWhere("post.price <= :upperBound", {upperBound: upperBound})
+      .where("post.original_price >= :lowerBound", {lowerBound: lowerBound})
+      .andWhere("post.original_price <= :upperBound", {upperBound: upperBound})
```
Comment (mateow99) [inline src/repositories/PostRepository.ts:99]: Wouldn't it be better to check if there is an altered price and use that and if not use the original price

File: src/services/UserService.ts [hunk 6]
```diff
@@ -18,7 +18,7 @@ export class UserService {
   }
 
   public async getAllUsers(user: UserModel): Promise<UserModel[]> {
-    if (!user.admin) throw new UnauthorizedError('User does not have permission to get all users')
+    // if (!user.admin) throw new UnauthorizedError('User does not have permission to get all users')
```
Comment (mateow99) [inline src/services/UserService.ts:21]: Why is this being commented out?

**Review by mateow99 — APPROVED** [review]
LGTM

**Review by joycrwu — APPROVED** [review]
lgtm

**Pull request merged**